torch/distributed/checkpoint/state_dict_loader.py

```diff
@@ -215,3 +216,66 @@ def read_data():
         return None
 
     _ = distW.all_gather("read", read_data)
+
+
+def _load_state_dict_from_keys(
+    keys: Optional[set],
```

Comment:
Why is this optional? If users pass None, is this API going to load everything? If so, we may want to mention that in the docstring. If this API will do nothing, then I suggest that we should simply not support the None case.